iceoryx_binding_c/test/moduletests/test_chunk.cpp

```diff
@@ -0,0 +1,169 @@
+    void* userPayload;
+    ASSERT_EQ(iox_pub_loan_chunk(publisher, &userPayload, USER_PAYLOAD_SIZE), AllocationResult_SUCCESS);
+
+    auto chunkHeader = iox_chunk_header_from_user_payload(userPayload);
+    ASSERT_NE(chunkHeader, nullptr);
+
+    // a default created ChunkHeader has always an adjacent user-payload
+    const uint64_t chunkStartAddress{reinterpret_cast<uint64_t>(chunkHeader)};
+    const uint64_t userPayloadStartAddress{reinterpret_cast<uint64_t>(userPayload)};
+    EXPECT_THAT(userPayloadStartAddress - chunkStartAddress, Eq(sizeof(ChunkHeader)));
+}
+
+TEST_F(BindingC_Chunk_test, GettingChunkHeaderFromConstUserPayloadWorks)
+{
+    iox_runtime_init("hypnotoad");
+
+    iox_pub_storage_t publisherStorage;
+    iox_pub_t publisher = iox_pub_init(&publisherStorage, "All", "Glory", "Hypnotoad", nullptr);
+
+    constexpr uint32_t USER_PAYLOAD_SIZE(42U);
+    void* userPayload;
+    ASSERT_EQ(iox_pub_loan_chunk(publisher, &userPayload, USER_PAYLOAD_SIZE), AllocationResult_SUCCESS);
+    const void* constUserPayload = userPayload;
+
+    auto chunkHeader = iox_chunk_header_from_user_payload_const(constUserPayload);
+    ASSERT_NE(chunkHeader, nullptr);
+
+    // a default created ChunkHeader has always an adjacent user-payload
+    const uint64_t chunkStartAddress{reinterpret_cast<uint64_t>(chunkHeader)};
+    const uint64_t userPayloadStartAddress{reinterpret_cast<uint64_t>(userPayload)};
+    EXPECT_THAT(userPayloadStartAddress - chunkStartAddress, Eq(sizeof(ChunkHeader)));
+}
+
+TEST_F(BindingC_Chunk_test, UserPayloadChunkHeaderUserPayloadRoundtripWorksForNonConst)
+{
+    iox_runtime_init("hypnotoad");
+
+    iox_pub_storage_t publisherStorage;
+    iox_pub_t publisher = iox_pub_init(&publisherStorage, "All", "Glory", "Hypnotoad", nullptr);
+
+    constexpr uint32_t USER_PAYLOAD_SIZE(42U);
+    void* userPayload;
+    ASSERT_EQ(iox_pub_loan_chunk(publisher, &userPayload, USER_PAYLOAD_SIZE), AllocationResult_SUCCESS);
+    const void* constUserPayload = userPayload;
+
+    auto chunkHeader = iox_chunk_header_from_user_payload_const(constUserPayload);
+    auto userPayloadRoundtrip = iox_chunk_header_to_user_payload_const(chunkHeader);
+
+    EXPECT_EQ(userPayloadRoundtrip, userPayload);
+}
+
+TEST_F(BindingC_Chunk_test, UserPayloadChunkHeaderUserPayloadRoundtripWorksForConst)
+{
+    iox_runtime_init("hypnotoad");
+
+    iox_pub_storage_t publisherStorage;
+    iox_pub_t publisher = iox_pub_init(&publisherStorage, "All", "Glory", "Hypnotoad", nullptr);
+
+    constexpr uint32_t USER_PAYLOAD_SIZE(42U);
+    void* userPayload;
+    ASSERT_EQ(iox_pub_loan_chunk(publisher, &userPayload, USER_PAYLOAD_SIZE), AllocationResult_SUCCESS);
+
+    auto chunkHeader = iox_chunk_header_from_user_payload(userPayload);
+    auto userPayloadRoundtrip = iox_chunk_header_to_user_payload(chunkHeader);
+
+    EXPECT_EQ(userPayloadRoundtrip, userPayload);
+}
+
+TEST_F(BindingC_Chunk_test, GettingUserHeaderFromNonConstChunkHeaderWorks)
```

Comment:
Nitpick: I try to avoid `NonConstFoo` and `ConstFoo`. I prefer `Foo` and `ConstFoo`

Reply:
I try to be explicit